cookbook/smartContracts.js

```diff
@@ -0,0 +1,425 @@
+
+// ```js
+// waits for transaction completion and parses the result
+// we use the transaction hash we got when broadcasting the transaction
+const parsedOutcome = controller.awaitCompletedExecute(transactionOnNetwork);
+const values = parsedOutcome.contracts.values
+// ```
+
+// ## **Calling a smart contract and sending tokens (transfer & execute)**
+// Additionally, if an endpoint requires a payment when called, we can send tokens to the contract while creating a smart contract call transaction. 
+// Both EGLD and ESDT tokens or a combination of both can be sent. This functionality is supported by both the controller and the factory.
+
+// ```js
+{ // md-ignore
+const filePath = path.join("src", "testdata", "testwallets", "alice.pem");
+const sender = await Account.newFromPem(filePath);
+
+// the developer is responsible for managing the nonce
+sender.nonce  = await entrypoint.recallAccountNonce(sender.address)
+
+// load the contract bytecode
+const codeBuffer = await promises.readFile("../contracts/adder.wasm");
+const code = Code.fromBuffer(codeBuffer);
+
+// load the abi file
+abi = await loadAbiRegistry("src/testdata/adder.abi.json")
+
+// get the smart contracts controller
+const entrypoint = new DevnetEntrypoint();
+const controller = entrypoint.createSmartContractController(abi);
+
+const contractAddress = Address.newFromBech32("erd1qqqqqqqqqqqqqpgq7cmfueefdqkjsnnjnwydw902v8pwjqy3d8ssd4meug");
+
+// For deploy arguments, use "TypedValue" objects if you haven't provided an ABI to the factory: // md-as-comment
+let args = [new U32Value(42)];
+// Or use simple, plain JavaScript values and objects if you have provided an ABI to the factory: // md-as-comment
+args = [42];
+
+// creating the transfers
+const firstToken = new Token({ identifier: "TEST-38f249", nonce: 10 });
+const firstTransfer = new TokenTransfer({ token: firstToken, amount: 1n });
+
+const secondToken = new Token({ identifier: "BAR-c80d29" });
+const secondTransfer = new TokenTransfer({ token: secondToken, amount: 10000000000000000000n });
+
+const transaction = await controller.createTransactionForExecute(
+  sender,
+  sender.getNonceThenIncrement(),
+  {
+    contract: contractAddress,
+    gasLimit: 5000000n,
+    function: "add",
+    arguments: args,
+    nativeTransferAmount: 1000000000000000000n,
+    tokenTransfers: [firstTransfer, secondTransfer]
+  },
+);
+
+// broadcasting the transaction
+const txHash = await entrypoint.sendTransaction(transaction);
+
+console.log(txHash);
+} // md-ignore
+// ```
+
+// ## **Calling a smart contract using the factory**
+// Let's create the same smart contract call transaction, but using the `factory`.
+
+// ```js
+{ // md-ignore
+const filePath = path.join("src", "testdata", "testwallets", "alice.pem");
+const sender = await Account.newFromPem(filePath);
+
+// the developer is responsible for managing the nonce
+sender.nonce  = await entrypoint.recallAccountNonce(sender.address)
+
+// load the contract bytecode
```

Comment:
Rendering issue, here and above / below:
 - https://github.com/multiversx/mx-sdk-js-examples/blob/TOOL-488-cookbook-v-14/cookbook/cookbook.md#calling-a-smart-contract-using-the-factory